jax/api.py

```diff
@@ -68,16 +68,14 @@
 
 AxisName = Any
 
-# This TypeVar is used below to express the fact that function call signatures
-# are invariant under the jit, vmap, and pmap transformations.
-# Specifically, we statically assert that the return type is invariant.
-# Until PEP-612 is implemented, we cannot express the same invariance for
-# function arguments.
-# Note that the return type annotations will generally not strictly hold
+# This TypeVar is used below to express the fact that function types
+# (i.e. call signatures) are invariant under the jit, vmap, and pmap
+# transformations.
+# Note that the function type annotations will generally not strictly hold
 # in JIT internals, as Tracer values are passed through the function.
 # Should this raise any type errors for the tracing code in future, we can disable
 # type checking in parts of the tracing code, or remove these annotations.
-T = TypeVar("T")
+F = TypeVar("F", Callable, Callable[..., Any])
```

Comment:
It looks like the [recommendation from mypy](https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators) is to use `bound`, e.g.,:
```suggestion
F = TypeVar("F", bound=Callable[..., Any])
```

I'm not entirely sure why that would be any different, but it might be worth a try...